Gems/Profiler/Code/Source/CpuProfiler.cpp

```diff
@@ -317,7 +318,7 @@ namespace Profiler
     // Gets called when region ends and all data is set
     void CpuTimingLocalStorage::AddCachedRegion(const CachedTimeRegion& timeRegionCached)
     {
-        if (auto iter = m_hitSizeLimitMap.find(timeRegionCached.m_groupRegionName.m_regionName);
+        if (auto iter = m_hitSizeLimitMap.find(timeRegionCached.m_groupRegionName.m_regionName.GetCStr());
```

Comment:
Because this is a map of `AZStd::string` the `m_regionName.GetStringView` function should be used instead to avoid retrieving a null-terminated c-string and needing needing the string constructor iterate over the string to determine it's string length again. The `AZStd::string` has a constructor which accepts a string_view that just reads the size directly from it.
```suggestion
        if (auto iter = m_hitSizeLimitMap.find(timeRegionCached.m_groupRegionName.m_regionName.GetStringView());
```